components/download_images/src/main.py

```diff
@@ -123,8 +124,9 @@ def transform(
             max_aspect_ratio=max_aspect_ratio,
         )
 
-        # Remove duplicates from laion retrieval
+        # Remove duplicates from laion retrieval (global)
         dataframe = dataframe.drop_duplicates()
+        dataframe = dataframe.repartition(npartitions=os.cpu_count())
```

Comment:
If we can consider both, that would be ideal, but not sure if we can.

Reply:
I have a new updated workflow that I think can mitigate the issue of having large data (also added it to the doc). That way we only repartition if we need but the reparitions should still be smaller than `RAM x os.cpu_count()`   
 
![image](https://github.com/ml6team/fondant/assets/47530815/e25a64c9-bdc1-4acc-91ae-089412832693)
 
 There is still the issue of having single partitions being larger than RAM, I proposed a way in which this can be tackled but would need to investigate more. Not sure if there is a clear-cut solution for that one.

Reply:
Can we actually know if `n_partitions<n_workers` at graph construction time?

We can make the first repartition based on memory dynamic. Where we use 250Mb by default, but the user can define a lower number if they know that their component explodes the memory usage per row.

Reply:
> Can we actually know if `n_partitions<n_workers` at graph construction time?

Yes, maybe it's not clear but this block would then be part of the transform component, i'v updated the plot. 


```python
df = dd.read_parquet()
n_partitions = df.npartitions
n_workers = os.cpu_count()
if n_partitions<n_workers:
     n_partitions = n_workers
df = df.repartition(n_partitions=n_partitions)
return df # dataframe returned to the user 
```


> We can make the first repartition based on memory dynamic. Where we use 250Mb by default, but the user can define a lower number if they know that their component explodes the memory usage per row.

I think this approach could work, but would involve quite a bit of trial and error since you probably will adjust it if your pipeline fails

Reply:
Ok, the split between the load and transform component was not clear to me before. I thought this was all happening in the transform component. I think this makes sense indeed!

Reply:
Can you update this PR to reflect this flow?